packages/transforms/src/fix-transform.ts

```diff
@@ -0,0 +1,57 @@
+import ts from 'typescript';
+
+import type { RehearsalService } from '@rehearsal/service';
+
+export type CodeFixAction = 'add' | 'delete';
+
+export interface FixedFile {
+  fileName: string;
+  updatedText: string;
+  code?: string;
+  codeFixAction?: CodeFixAction;
+  location: {
+    line: number;
+    character: number;
+  };
+}
+
+export interface FixTransformI {
+  /** Engineer friendly message describes steps needs to be done to fix the related issue */
+  hint?: string;
+
+  /** Function to fix the diagnostic issue */
+  fix?: (diagnostic: ts.DiagnosticWithLocation, service: RehearsalService) => FixedFile[];
+
+  run(diagnostic: ts.DiagnosticWithLocation, service: RehearsalService): FixedFile[];
+}
+
+export class FixTransform implements FixTransformI {
```

Comment:
We're planning to rename this to `Codefix` (`Codefix1234`), but probably it's better to have separate PR for this :)